cmd/root.go

```diff
@@ -138,10 +165,29 @@ func (a *MethodK8s) InitRootCommand() {
 
 	a.RootCmd.PersistentFlags().BoolVarP(&a.RootFlags.Quiet, "quiet", "q", false, "Suppress output")
 	a.RootCmd.PersistentFlags().BoolVarP(&a.RootFlags.Verbose, "verbose", "v", false, "Verbose output")
-	a.RootCmd.PersistentFlags().StringVarP(&a.RootFlags.Context, "context", "c", "", "Name of Context you want to use (ie. minikube)")
-	a.RootCmd.PersistentFlags().StringVarP(&a.RootFlags.Path, "path", "p", "", "Absolute or relative path to the Config file (ie. ~/.kube/config)")
-	a.RootCmd.PersistentFlags().StringVarP(&a.RootFlags.URL, "url", "u", "", "Cluster URL (ie. mycluster.com)")
-	a.RootCmd.PersistentFlags().BoolVarP(&a.RootFlags.Secret, "secret", "s", false, "Set to true if you want to use a token, CA, and URL to authenticate")
+
+	a.RootCmd.PersistentFlags().BoolVarP(&a.RootFlags.ServiceAccount, "service-account", "s", false, "Set to true if using service account workflow")
+	a.RootCmd.PersistentFlags().StringVarP(&a.RootFlags.Token, "token", "t", "", "Base64 encoded service account token")
+	a.RootCmd.PersistentFlags().StringVarP(&a.RootFlags.CACert, "cert", "a", "", "Base64 encoded ca certificate")
+	a.RootCmd.PersistentFlags().StringVarP(&a.RootFlags.APIServerURL, "server-url", "e", "", "Cluster server url")
+
+	a.RootCmd.PersistentFlags().StringVarP(&a.RootFlags.Context, "context", "c", "", "Cluster context (ie. minikube)")
+	a.RootCmd.PersistentFlags().StringVarP(&a.RootFlags.Path, "path", "p", "", "Absolute or relative path to the config file (ie. ~/.kube/config)")
+	a.RootCmd.PersistentFlags().StringVarP(&a.RootFlags.URL, "url", "u", "", "Cluster url (ie. mycluster.com)")
+
+	a.RootCmd.MarkFlagsMutuallyExclusive("context", "service-account")
+	a.RootCmd.MarkFlagsMutuallyExclusive("context", "token")
+	a.RootCmd.MarkFlagsMutuallyExclusive("context", "cert")
+	a.RootCmd.MarkFlagsMutuallyExclusive("context", "server-url")
+	a.RootCmd.MarkFlagsMutuallyExclusive("path", "service-account")
+	a.RootCmd.MarkFlagsMutuallyExclusive("path", "token")
+	a.RootCmd.MarkFlagsMutuallyExclusive("path", "cert")
+	a.RootCmd.MarkFlagsMutuallyExclusive("path", "server-url")
+	a.RootCmd.MarkFlagsMutuallyExclusive("url", "service-account")
+	a.RootCmd.MarkFlagsMutuallyExclusive("url", "token")
+	a.RootCmd.MarkFlagsMutuallyExclusive("url", "cert")
+	a.RootCmd.MarkFlagsMutuallyExclusive("url", "server-url")
+
```

Comment:
No there are for sure common pairings but no strict groups. Ie. `--context` and `--path` will be run together alot but you can still run one without the other. Even just `--context` without the `--path` can be used because the CLI will then check for a value in`$KUBECONFIG`